From fb2d418327bf152b9fd87424d44a7909db1e9330 Mon Sep 17 00:00:00 2001
From: Stefano Pigozzi <stefano.pigozzi@gmail.com>
Date: Thu, 2 Jan 2014 20:34:14 +0100
Subject: [PATCH] build: fix cocoa configure check on OS X 10.7

It failed because the 10.7 SDK doesn't natively support array and dictionary
subscripting.
---
 waftools/checks/custom.py  | 13 ++++++++++++-
 waftools/fragments/cocoa.m |  2 ++
 wscript                    |  6 +-----
 3 files changed, 15 insertions(+), 6 deletions(-)

diff --git a/waftools/checks/custom.py b/waftools/checks/custom.py
index 1b698ec..16f876e 100644
--- a/waftools/checks/custom.py
+++ b/waftools/checks/custom.py
@@ -3,7 +3,8 @@ from waftools.checks.generic import *
 from waflib import Utils
 import os
 
-__all__ = ["check_pthreads", "check_iconv", "check_lua", "check_oss_4front"]
+__all__ = ["check_pthreads", "check_iconv", "check_lua", "check_oss_4front",
+           "check_cocoa"]
 
 pthreads_program = load_fragment('pthreads.c')
 
@@ -101,3 +102,13 @@ def check_oss_4front(ctx, dependency_identifier):
                   fragment=load_fragment('oss_audio.c'))
 
     return fn(ctx, dependency_identifier)
+
+def check_cocoa(ctx, dependency_identifier):
+    fn = check_cc(
+        fragment         = load_fragment('cocoa.m'),
+        compile_filename = 'test.m',
+        framework_name   = ['Cocoa', 'IOKit', 'OpenGL'],
+        includes         = ctx.srcnode.abspath(),
+        linkflags        = '-fobjc-arc')
+
+    return fn(ctx, dependency_identifier)
diff --git a/waftools/fragments/cocoa.m b/waftools/fragments/cocoa.m
index 6e0e4d3..8096afb 100644
--- a/waftools/fragments/cocoa.m
+++ b/waftools/fragments/cocoa.m
@@ -2,6 +2,8 @@
 #import <OpenGL/OpenGL.h>
 #import <Cocoa/Cocoa.h>
 
+#include "osdep/macosx_compat.h"
+
 int main(int argc, char **argv) {
     @autoreleasepool {
         NSArray *ary = @[@1, @2, @3];
diff --git a/wscript b/wscript
index e6e0a83..eda8d4f 100644
--- a/wscript
+++ b/wscript
@@ -498,11 +498,7 @@ video_output_features = [
     {
         'name': '--cocoa',
         'desc': 'Cocoa',
-        'func': check_cc(
-            fragment=load_fragment('cocoa.m'),
-            compile_filename='test.m',
-            framework_name=['Cocoa', 'IOKit', 'OpenGL'],
-            linkflags='-fobjc-arc')
+        'func': check_cocoa
     } , {
         'name': 'gdi',
         'desc': 'GDI',
-- 
1.7.11.1

From d4f37174a16afad6394e48e203abb41252e382d2 Mon Sep 17 00:00:00 2001
From: Stefano Pigozzi <stefano.pigozzi@gmail.com>
Date: Thu, 2 Jan 2014 22:42:16 +0100
Subject: [PATCH] vda: fix build on OS X 10.7

Looks like on 10.8 OpenGL.h recursively includes CGLIOSurface.h. That is not
the case for 10.7 so the build was broken on that version of OS X.
---
 video/out/gl_hwdec_vda.c | 1 +
 video/out/vo_corevideo.c | 1 +
 2 files changed, 2 insertions(+)

diff --git a/video/out/gl_hwdec_vda.c b/video/out/gl_hwdec_vda.c
index 945a57b..c86c1ab 100644
--- a/video/out/gl_hwdec_vda.c
+++ b/video/out/gl_hwdec_vda.c
@@ -20,6 +20,7 @@
 #include <IOSurface/IOSurface.h>
 #include <CoreVideo/CoreVideo.h>
 #include <OpenGL/OpenGL.h>
+#include <OpenGL/CGLIOSurface.h>
 
 #include "video/decode/dec_video.h"
 #include "cocoa_common.h"
diff --git a/video/out/vo_corevideo.c b/video/out/vo_corevideo.c
index 92b3c76..20769d7 100644
--- a/video/out/vo_corevideo.c
+++ b/video/out/vo_corevideo.c
@@ -25,6 +25,7 @@
 #include <QuartzCore/QuartzCore.h>
 #if HAVE_VDA_HWACCEL
 #include <IOSurface/IOSurface.h>
+#include <OpenGL/CGLIOSurface.h>
 #endif
 
 #include <assert.h>
-- 
1.7.11.1

